clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -552,13 +553,18 @@ public void handle(OffsetCommitResponse commitResponse, RequestFuture<Void> futu
                         || error == Errors.ILLEGAL_GENERATION
                         || error == Errors.REBALANCE_IN_PROGRESS) {
                     // need to re-join group
-                    log.error("Error {} occurred while committing offsets for group {}", error, groupId);
+                    log.info("Offset commit for group {} failed due to {}, will find new coordinator and retry", groupId, error);
                     subscriptions.needReassignment();
-                    future.raise(new CommitFailedException("Commit cannot be completed due to group rebalance"));
+                    future.raise(new CommitFailedException("Commit cannot be completed since the group has already " +
+                            "rebalanced and assigned the partitions to another member. This most likely means that " +
```

Comment:
Do we want to say "most likely means" or simply "may mean"? There could be other reasons too, right?